File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.java
class Solution {
    public int findMin(int[] nums) {
        int l = 0;
        int h = nums.length-1;
        if (nums[l] <= nums[h]) return nums[l];
        while(l<=h) {
            int m = (l+h)/2;
            if (m > 0 && nums[m] < nums[m-1]) {
                return nums[m];
            }
            if (nums[m] > nums[h]) {
                l = m+1;
            } else {
                h = m-1;
            }
        }
        return nums[l];
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        //store the count in map
        //key:value = number:count
        unordered_map<int,int> freq;
        for(int i = 0; i < nums.size(); i++){
            freq[nums[i]] += 1;
        }
        //create a list of{count, number}
        vector<pair<int,int>> count;
        for(const auto& p : freq){
            count.push_back({p.second,p.first});
        }
        //{1, 1},{2,2},{3,3}
        //sort on count in reverse order
        sort(count.rbegin(),count.rend());
        //add to a result array the top 2 elements
        vector<int> res;
        for(int i = 0; i < k; i++){
            res.push_back(count[i].second);
        }
        return res;
       
    }
};